arviz/stats/stats.py

```diff
@@ -2043,3 +2044,61 @@ def apply_test_function(
         setattr(out, grp, out_group)
 
     return out
+
+
+def weight_predictions(idatas, weights=None):
+    """
+    Generate weighted posterior predictive samples from a list of InferenceData
+    and a set of weights.
+
+    Parameters
+    ---------
+    datasets: list[InfereneData]
+        List of :class:`arviz.InferenceData` objects containing the groups `posterior_predictive`
+        and `observed_data`. Observations should be the same for all InferenceData objects.
+    weights : array-like, optional
+        Individual weights for each model. Weights should be positive. If they do not sum up to 1,
+        they will be normalized. Default, same weight for each model.
+        Weights can be computed using many different methods including those in
+        :func:`arviz.compare`.
+
+    Returns
+    -------
+    idata: InferenceData
+        Output InferenceData object with the groups `posterior_predictive` and `observed_data`.
+
+    See Also
+    --------
+    compare :  Compare models based on PSIS-LOO `loo` or WAIC `waic` cross-validation
+    """
+    if weights is None:
+        weights = np.ones(len(idatas)) / len(idatas)
+    weights = np.array(weights, dtype=float)
+    weights /= weights.sum()
+
+    if not np.all(["posterior_predictive" in idata.groups() for idata in idatas]):
+        raise ValueError(
+            "All the InferenceData objects must contain the `posterior_predictive` group"
+        )
+
+    if not np.all([idatas[0].observed_data.equals(idata.observed_data) for idata in idatas[1:]]):
```

Comment:
```suggestion
    if not all(idatas[0].observed_data.equals(idata.observed_data) for idata in idatas[1:]):
```

Also need to handle cases with 1 or 0 idatas len